VPLAY-10563 curl timeout disambiguation

lashmintha opened the pull request:
Reason for change: Distinguished the curl error 28, into DNS timeout error, connection with the host timeout and data transfer timeout
Risks: Low
Test Procedure: Refer jira ticket
Priority: P1

lashmintha committed:
VPLAY-10563 curl timeout disambiguation

Reason for change: Distinguished the curl error 28, into DNS timeout error, connection with the host timeout and data transfer timeout
Risks: Low
Test Procedure: Refer jira ticket
Priority: P1

Signed-off-by: lashmintha <lashminthaj17@gmail.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=576) <br/>All committers have signed the CLA.

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
74ea4cbc62440d36c61b4ff53bc2f4af
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"headerValue" is passed-by-value as parameter to "std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::vector(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""headerValue"")" instead of "headerValue".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:6086](undefined//blob/undefined/priv_aamp.cpp#L6086)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d06859e9a18c34edab98a5d07134d57f
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"aesDecPtr" is passed-by-value as parameter to "std::shared_ptr<AesDec>::shared_ptr(std::shared_ptr<AesDec> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""aesDecPtr"")" instead of "aesDecPtr".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/DrmInterface.cpp:215](undefined//blob/undefined/drm/DrmInterface.cpp#L215)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
412105afd9b3cd3924f653a5098c25a7
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"fragmentUrl" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""fragmentUrl"")" instead of "fragmentUrl".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:6836](undefined//blob/undefined/priv_aamp.cpp#L6836)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
159ef0da2295896715f72414b88d2207
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"eventHandle" is passed-by-value as parameter to "std::shared_ptr<DrmMetaDataEvent>::shared_ptr(std::shared_ptr<DrmMetaDataEvent> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""eventHandle"")" instead of "eventHandle".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:1315](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L1315)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b1e795ed66ce3bfc421e796835b0e465
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:175](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L175)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
5501f08eb8a6fce118899dfe1cc84ddf
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:923](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L923)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
a06a220113967426dab306e1871f3dc7
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"remoteUrl" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""remoteUrl"")" instead of "remoteUrl".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[MediaStreamContext.cpp:488](undefined//blob/undefined/MediaStreamContext.cpp#L488)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
6dd4613f444a55c22769ef51dcdf5a79
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inpData" is passed-by-value as parameter to "std::shared_ptr<_manifestDownloadConfig>::shared_ptr(std::shared_ptr<_manifestDownloadConfig> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inpData"")" instead of "inpData".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:5286](undefined//blob/undefined/priv_aamp.cpp#L5286)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
6ee8c3fe344453a037ff3bf15a371cbb
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:2907](undefined//blob/undefined/priv_aamp.cpp#L2907)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
21d8ba029d164c422e04c391f71da005
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"fragmentUrl" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""fragmentUrl"")" instead of "fragmentUrl".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[MediaStreamContext.cpp:411](undefined//blob/undefined/MediaStreamContext.cpp#L411)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
8d87d7af21450c54e8ece7a950c80278
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"fragmentToTsbSessionMgr" is passed-by-value as parameter to "std::shared_ptr<CachedFragment>::shared_ptr(std::shared_ptr<CachedFragment> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""fragmentToTsbSessionMgr"")" instead of "fragmentToTsbSessionMgr".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[MediaStreamContext.cpp:428](undefined//blob/undefined/MediaStreamContext.cpp#L428)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
ab80de788d16a67ac4842925063269c8
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"curInstreamId" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""curInstreamId"")" instead of "curInstreamId".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:12371](undefined//blob/undefined/priv_aamp.cpp#L12371)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
544e0c5fc51f81758ed82f9e5a7b6fe6
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"sTraceId" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""sTraceId"")" instead of "sTraceId".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:6259](undefined//blob/undefined/priv_aamp.cpp#L6259)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
1a3971b71c5a7cdc222e8f276e6e8e3f
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"processor" is passed-by-value as parameter to "std::function<void (AampMediaType, SegmentInfo_t, std::vector<unsigned char, std::allocator<unsigned char> >)>::function(std::function<void (AampMediaType, SegmentInfo_t, std::vector<unsigned char, std::allocator<unsigned char> >)> const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""processor"")" instead of "processor".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[MediaStreamContext.cpp:43](undefined//blob/undefined/MediaStreamContext.cpp#L43)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
3ab3b3c684fea8fc364337e025c0f3d5
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputNameString" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputNameString"")" instead of "inputNameString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:11671](undefined//blob/undefined/priv_aamp.cpp#L11671)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
dff93acca738246fe317786a72a8bb1b
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputRenditionString" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputRenditionString"")" instead of "inputRenditionString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:11664](undefined//blob/undefined/priv_aamp.cpp#L11664)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
82e34b26df8b71a7fefbbc6c6f204e60
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inpData" is passed-by-value as parameter to "std::shared_ptr<_downloadConfig>::shared_ptr(std::shared_ptr<_downloadConfig> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inpData"")" instead of "inpData".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:13329](undefined//blob/undefined/priv_aamp.cpp#L13329)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
4d0a6b7e8638099e5518a7d2d75a94bd
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"value" is copied and then passed-by-reference as parameter to STL insertion function "std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::push_back(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::value_type const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""value"")" instead of "value".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:13547](undefined//blob/undefined/priv_aamp.cpp#L13547)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
4990f51541689e3cdd347dffe488959e
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inpData" is passed-by-value as parameter to "std::shared_ptr<_downloadConfig>::shared_ptr(std::shared_ptr<_downloadConfig> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inpData"")" instead of "inpData".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:4772](undefined//blob/undefined/priv_aamp.cpp#L4772)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
27cef176fa117c014eda23352d83b928
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"licenseServerProxy" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""licenseServerProxy"")" instead of "licenseServerProxy".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:385](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L385)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
1ac4c876dd4a91b7abb5d28c4e963007
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inpData" is passed-by-value as parameter to "std::shared_ptr<_downloadConfig>::shared_ptr(std::shared_ptr<_downloadConfig> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inpData"")" instead of "inpData".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:2802](undefined//blob/undefined/priv_aamp.cpp#L2802)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d3e6446538c70e3dc8d617c75278623b
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputLanguagesList" is copied in call to copy assignment for class "std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputLanguagesList"")" instead of "inputLanguagesList".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:11667](undefined//blob/undefined/priv_aamp.cpp#L11667)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
2f3f1e493060d7a5ff800c05f74cca9c
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputTextRenditionString" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputTextRenditionString"")" instead of "inputTextRenditionString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:12237](undefined//blob/undefined/priv_aamp.cpp#L12237)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
8e6d78b8f6f467762155b76c181bc6ea
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"temp" is copied and then passed-by-reference as parameter to STL insertion function "std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::push_back(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::value_type const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""temp"")" instead of "temp".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:768](undefined//blob/undefined/priv_aamp.cpp#L768)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
01464f1d61eb6535588bb311a7315ba4
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputLanguagesString" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputLanguagesString"")" instead of "inputLanguagesString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:11668](undefined//blob/undefined/priv_aamp.cpp#L11668)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
38c183a19a6a09a50711e05779aefeb1
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"fragmentToCache" is passed-by-value as parameter to "std::shared_ptr<CachedFragment>::shared_ptr(std::shared_ptr<CachedFragment> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""fragmentToCache"")" instead of "fragmentToCache".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[MediaStreamContext.cpp:450](undefined//blob/undefined/MediaStreamContext.cpp#L450)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
ca113e1c7cacd1a481357cc5ea91798e
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:1420](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L1420)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
2d8969a0f7cd176f6b3aecfe4d9cefe9
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputTextLanguagesList" is copied in call to copy assignment for class "std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputTextLanguagesList"")" instead of "inputTextLanguagesList".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:12266](undefined//blob/undefined/priv_aamp.cpp#L12266)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
bbaff512db5f06ecd60d5584f60845d2
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"sCustomHeaders" is copied in call to copy assignment for class "std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > >", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""sCustomHeaders"")" instead of "sCustomHeaders".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:13550](undefined//blob/undefined/priv_aamp.cpp#L13550)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d6a377658a11a676e69432f0f5c33ee0
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"selectedTextTrack" is passed-by-value as parameter to "TextTrackInfo::TextTrackInfo(TextTrackInfo const &) /*implicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""selectedTextTrack"")" instead of "selectedTextTrack".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:12426](undefined//blob/undefined/priv_aamp.cpp#L12426)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b4197e87a86c37be91d7086c63147c25
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputLabelsString" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputLabelsString"")" instead of "inputLabelsString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:11666](undefined//blob/undefined/priv_aamp.cpp#L11666)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
5f104fcf1e3d3db32070392c09752c95
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"effectiveUrl" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""effectiveUrl"")" instead of "effectiveUrl".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[streamabstraction.cpp:4393](undefined//blob/undefined/streamabstraction.cpp#L4393)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
9dfd34130af46dfe43199ea10c0581e2
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inpData" is passed-by-value as parameter to "std::shared_ptr<_manifestDownloadConfig>::shared_ptr(std::shared_ptr<_manifestDownloadConfig> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inpData"")" instead of "inpData".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:5756](undefined//blob/undefined/priv_aamp.cpp#L5756)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
3dc5b5a066058aeb8240d4bf18a7ab57
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputTextTypeString" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputTextTypeString"")" instead of "inputTextTypeString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:12240](undefined//blob/undefined/priv_aamp.cpp#L12240)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
415a3ec8eef90538154ba92ab59dfa8b
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"modUrl" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""modUrl"")" instead of "modUrl".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:6714](undefined//blob/undefined/priv_aamp.cpp#L6714)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d9b729cf93f08d440255ad56563dd034
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"mID3Handler" is passed-by-value as parameter to "std::function<void (AampMediaType, unsigned char const *, unsigned long, SegmentInfo_t const &, char const *)>::function(std::function<void (AampMediaType, unsigned char const *, unsigned long, SegmentInfo_t const &, char const *)> const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""mID3Handler"")" instead of "mID3Handler".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[streamabstraction.cpp:2157](undefined//blob/undefined/streamabstraction.cpp#L2157)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
dcfc848b6a72a673c8955c041ca54f44
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputCodecList" is copied in call to copy assignment for class "std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputCodecList"")" instead of "inputCodecList".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:11670](undefined//blob/undefined/priv_aamp.cpp#L11670)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
791cb847561419d6ecd2837e4c8bdb43
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputTextLabelString" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputTextLabelString"")" instead of "inputTextLabelString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:12239](undefined//blob/undefined/priv_aamp.cpp#L12239)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
f1a8bcd2e1db54de2304a01c111804a7
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputTextLanguagesString" is copied and then passed-by-reference as parameter to STL insertion function "std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::push_back(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::value_type const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputTextLanguagesString"")" instead of "inputTextLanguagesString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:12158](undefined//blob/undefined/priv_aamp.cpp#L12158)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
8e9cf64c3606003cefa12d8c2cfdfac7
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputTextNameString" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputTextNameString"")" instead of "inputTextNameString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:12242](undefined//blob/undefined/priv_aamp.cpp#L12242)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
262a33c46614ce66a8f6a2937f4ea056
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"headerName" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""headerName"")" instead of "headerName".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:6086](undefined//blob/undefined/priv_aamp.cpp#L6086)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
5f104fcf1e3d3db32070392c09752c95
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"effectiveUrl" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""effectiveUrl"")" instead of "effectiveUrl".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[streamabstraction.cpp:4399](undefined//blob/undefined/streamabstraction.cpp#L4399)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e6a4946a5fe7380b173a40320168e778
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"keyId" is passed-by-value as parameter to "std::vector<unsigned char, std::allocator<unsigned char> >::vector(std::vector<unsigned char, std::allocator<unsigned char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""keyId"")" instead of "keyId".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:1429](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L1429)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
699ef876b17b79defc615402bbdc57f2
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"url" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""url"")" instead of "url".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[MediaStreamContext.cpp:689](undefined//blob/undefined/MediaStreamContext.cpp#L689)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
cb820765b6010c4efd1247df0d231946
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"id" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""id"")" instead of "id".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:8062](undefined//blob/undefined/priv_aamp.cpp#L8062)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
8d87d7af21450c54e8ece7a950c80278
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"fragmentToTsbSessionMgr" is passed-by-value as parameter to "std::shared_ptr<CachedFragment>::shared_ptr(std::shared_ptr<CachedFragment> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""fragmentToTsbSessionMgr"")" instead of "fragmentToTsbSessionMgr".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[MediaStreamContext.cpp:411](undefined//blob/undefined/MediaStreamContext.cpp#L411)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d2924abf7c2fbe728317d334a0bf8ea1
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"headerValue" is copied in call to copy assignment for class "std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""headerValue"")" instead of "headerValue".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:8915](undefined//blob/undefined/priv_aamp.cpp#L8915)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
4e4d3e6355a13f99ac2107ffc74274f0
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputLabelList" is copied in call to copy assignment for class "std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputLabelList"")" instead of "inputLabelList".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:11665](undefined//blob/undefined/priv_aamp.cpp#L11665)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d2924abf7c2fbe728317d334a0bf8ea1
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"headerValue" is copied in call to copy assignment for class "std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""headerValue"")" instead of "headerValue".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:8930](undefined//blob/undefined/priv_aamp.cpp#L8930)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
1fe564debd9ade9557b95bcc8b156d00
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputCodecString" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputCodecString"")" instead of "inputCodecString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:11669](undefined//blob/undefined/priv_aamp.cpp#L11669)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e056628e402b32c98cbcf32de91bcb25
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"ccTrack" is copied and then passed-by-reference as parameter to STL insertion function "std::vector<CCTrackInfo, std::allocator<CCTrackInfo> >::push_back(std::vector<CCTrackInfo, std::allocator<CCTrackInfo> >::value_type const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""ccTrack"")" instead of "ccTrack".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:519](undefined//blob/undefined/priv_aamp.cpp#L519)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
6dd4613f444a55c22769ef51dcdf5a79
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inpData" is passed-by-value as parameter to "std::shared_ptr<_manifestDownloadConfig>::shared_ptr(std::shared_ptr<_manifestDownloadConfig> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inpData"")" instead of "inpData".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:5290](undefined//blob/undefined/priv_aamp.cpp#L5290)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
075ca3394b4031b17f2eb7f20041d132
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"jsonStr" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""jsonStr"")" instead of "jsonStr".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:13328](undefined//blob/undefined/priv_aamp.cpp#L13328)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
f1a8bcd2e1db54de2304a01c111804a7
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputTextLanguagesString" is copied and then passed-by-reference as parameter to STL insertion function "std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::push_back(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::value_type const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputTextLanguagesString"")" instead of "inputTextLanguagesString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:12151](undefined//blob/undefined/priv_aamp.cpp#L12151)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b647423171a4a05e7dbec5d975b3964c
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"periodId" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""periodId"")" instead of "periodId".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:923](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L923)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b521d65ac45f27f1d82b54abc8f45598
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"inputInstreamIdString" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inputInstreamIdString"")" instead of "inputInstreamIdString".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:12241](undefined//blob/undefined/priv_aamp.cpp#L12241)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e1994e214f09eafa25196786d0c7be36
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"respData" is passed-by-value as parameter to "std::shared_ptr<_downloadResponse>::shared_ptr(std::shared_ptr<_downloadResponse> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""respData"")" instead of "respData".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:2803](undefined//blob/undefined/priv_aamp.cpp#L2803)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
cb820765b6010c4efd1247df0d231946
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"id" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""id"")" instead of "id".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:8058](undefined//blob/undefined/priv_aamp.cpp#L8058)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
eddd4cbd2433a72cb427ad12ba7df325
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"licenseProxy" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""licenseProxy"")" instead of "licenseProxy".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:945](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L945)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
bdc12a67d1df7ea39c2a7787dd623564
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"track" is passed-by-value as parameter to "TextTrackInfo::TextTrackInfo(TextTrackInfo const &) /*implicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""track"")" instead of "track".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:10983](undefined//blob/undefined/priv_aamp.cpp#L10983)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
24ab28a5cf591599992d7509310e7659
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"event" is passed-by-value as parameter to "std::shared_ptr<AAMPEventObject>::shared_ptr(std::shared_ptr<AAMPEventObject> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""event"")" instead of "event".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:2960](undefined//blob/undefined/priv_aamp.cpp#L2960)

rdkcmf-jenkins commented on drm/AampDRMLicManager.cpp:
```diff
@@ -640,60 +640,52 @@ const char * AampDRMLicenseManager::getAccessToken(int &tokenLen, int &error_cod
 		inpData->bSSLVerifyPeer		=	bSslPeerVerify;
 		mAccessTokenConnector.Initialize(inpData);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
93e84ecebb6d3cc407b05739a838c523
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Variable copied when it could be moved
"inpData" is passed-by-value as parameter to "std::shared_ptr<_downloadConfig>::shared_ptr(std::shared_ptr<_downloadConfig> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""inpData"")" instead of "inpData".

</details>

rdkcmf-jenkins commented on drm/AampDRMLicManager.cpp:
```diff
@@ -640,60 +640,52 @@ const char * AampDRMLicenseManager::getAccessToken(int &tokenLen, int &error_cod
+			}
+			if(tokenStatusCode.length() == 1 && tokenStatusCode.c_str()[0] == '0')
+			{
+				string token = extractSubstring(tokenReplyStr, "token\":\"", "\"");
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
7c0550d2d1fd28f9155203b361c07bcb
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"tokenReplyStr" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""tokenReplyStr"")" instead of "tokenReplyStr".

Vinish100 replied:
Please fix this warning

lashmintha replied:
Keeping as it is

Vinish100 commented on AampLogManager.h:
```diff
@@ -33,6 +33,8 @@
 #include <iomanip> // std::setfill
 #include <sstream> // std::ostringstream
 #include <algorithm> // std::foreach
+#include "AampCurlDownloader.h"
```
Is this header file required?

Vinish100 commented on AampMPDDownloader.cpp:
```diff
@@ -468,9 +465,18 @@ void AampMPDDownloader::downloadMPDThread1()
 
 			mMPDData->mMPDStatus	=	AAMPStatusType::eAAMPSTATUS_MANIFEST_DOWNLOAD_ERROR;
+			
 			if(mMPDData->mMPDDownloadResponse->iHttpRetValue != 200 && mMPDData->mMPDDownloadResponse->iHttpRetValue != 204 && mMPDData->mMPDDownloadResponse->iHttpRetValue != 206)
```
Can this be !IS_HTTP_SUCCESS(mMPDData->mMPDDownloadResponse->iHttpRetValue)?

Vinish100 commented on AampMPDDownloader.cpp:
```diff
@@ -521,8 +527,7 @@ void AampMPDDownloader::downloadMPDThread1()
 {
-	if(mMPDData->mMPDDownloadResponse->curlRetValue == 0 && 
-		(mMPDData->mMPDDownloadResponse->iHttpRetValue == 200 || mMPDData->mMPDDownloadResponse->iHttpRetValue == 206))
+	if( mMPDData->mMPDDownloadResponse->iHttpRetValue == 200 || mMPDData->mMPDDownloadResponse->iHttpRetValue == 206 )
```
Can we use IS_HTTP_SUCCESS?

Vinish100 commented on AampMPDDownloader.cpp:
```diff
@@ -584,7 +589,7 @@ void AampMPDDownloader::stichToCachedManifest(ManifestDownloadResponsePtr mpdToA
 void AampMPDDownloader::showDownloadMetrics(DownloadResponsePtr dnldPtr, int totalPerformanceTime)
 {
-	CURLcode res 			=	static_cast<CURLcode>(dnldPtr->curlRetValue);
+	//CURLcode res 			=	static_cast<CURLcode>(dnldPtr->curlRetValue);
```
Remove commented code

Vinish100 commented on AampMPDDownloader.cpp:
```diff
@@ -682,6 +696,25 @@ ManifestDownloadResponsePtr AampMPDDownloader::GetManifest(bool bWait, int iWait
 				// Timed out
 				respPtr->mMPDDownloadResponse->iHttpRetValue = CURLE_OPERATION_TIMEDOUT;
+
+				CURL *curlHandle = nullptr;
```
Initialize directly to mDownloader1.GetCurlHandle()

Vinish100 commented on AampMPDDownloader.h:
```diff
@@ -63,6 +63,7 @@
 #include "dash/mpd/MPDSegmenter.h"
 #include "AampLLDASHData.h"
 #include "AampMPDUtils.h"
+#include "AampUtils.h"
```
Required? If its for a function, please add it to .cpp

Vinish100 commented on AampUtils.cpp:
```diff
@@ -414,18 +413,9 @@ std::string aamp_PostJsonRPC( std::string id, std::string method, std::string pa
-		AAMPLOG_ERR("failed: %d", respData->curlRetValue);
-	}
-
+	AAMPLOG_WARN("JSONRPC data: %s", inpData->postData.c_str() );
```
Why this change?

Vinish100 commented on AampUtils.cpp:
```diff
@@ -1449,4 +1432,35 @@ bool aamp_isTuneScheme( const char *cmdBuf )
+}
+bool GetCurlTimeoutFailureStatus( int httpResponseCode )
+{
+	if( httpResponseCode == CURLE_OPERATION_TIMEDOUT || httpResponseCode == eCURL_TIMEOUT_DNS || httpResponseCode == eCURL_TIMEOUT_CONNECT || httpResponseCode == eCURL_TIMEOUT_DATA )
```
return (httpResponseCode == CURLE_OPERATION_TIMEDOUT || httpResponseCode == eCURL_TIMEOUT_DNS || httpResponseCode == eCURL_TIMEOUT_CONNECT || httpResponseCode == eCURL_TIMEOUT_DATA)

Vinish100 commented on downloader/AampCurlDownloader.cpp:
```diff
@@ -159,7 +158,7 @@ bool AampCurlDownloader::IsDownloadActive()
 {
 	int httpRetVal=0;
-	int curlRetVal=0;
+	//int curlRetVal=0;
```
Remove commented code

Vinish100 commented on downloader/AampCurlDownloader.cpp:
```diff
@@ -236,7 +235,10 @@ int AampCurlDownloader::Download(const std::string &urlStr, std::shared_ptr<Down
 						//Attempt retry for partial downloads, which have a higher chance to succeed
-						if (curlRetVal == CURLE_COULDNT_CONNECT || curlRetVal == CURLE_OPERATION_TIMEDOUT || curlRetVal  == CURLE_PARTIAL_FILE)
+						if (httpRetVal == CURLE_COULDNT_CONNECT ||
+						    GetCurlTimeoutFailureStatus (httpRetVal) ||
```
eCURL_TIMEOUT_CONNECT and eCURL_TIMEOUT_DATA is already checked in GetCurlTimeoutFailureStatus

Vinish100 commented on downloader/AampCurlDownloader.h:
```diff
@@ -176,8 +175,9 @@ typedef struct _downloadResponse
 	std::string sEffectiveUrl;
 	std::vector<std::string>  mResponseHeader;
 	std::vector<std::uint8_t> mDownloadData;
+	//CurlTimeoutFailureReason mCurlTimeoutFailureReason;
```
Remove commented out code

Vinish100 commented on fragmentcollector_mpd.cpp:
```diff
@@ -4521,6 +4521,7 @@ AAMPStatusType StreamAbstractionAAMP_MPD::FetchDashManifest()
 	double downloadTime;
 	bool updateVideoEndMetrics = false;
 	int http_error = 0;
+	//CurlTimeoutFailureReason failureReason;
```
Remove commented out code

Vinish100 commented on priv_aamp.cpp:
```diff
@@ -2655,6 +2658,14 @@ void PrivateInstanceAAMP::SendDownloadErrorEvent(AAMPTuneFailure tuneFailure, in
 		{
 			snprintf(description,MAX_DESCRIPTION_SIZE, "%s : Curl Error Code %d", tuneFailureMap[tuneFailure].description, error_code);  //CID:86441 - DC>STRING_BUFFER
 		}
+		else if( error_code >= 1000 )
```
Can be clubbed with else if(error_code < 100). Why separate block is required?

Vinish100 commented on priv_aamp.cpp:
```diff
@@ -4284,12 +4294,34 @@ bool PrivateInstanceAAMP::GetFile( std::string remoteUrl, AampMediaType mediaTyp
-						AampLogManager::LogNetworkError (effectiveUrl.c_str(), // Effective URL could be different than remoteURL
-						AAMPNetworkErrorCurl, (int)(progressCtx.abortReason == eCURL_ABORT_REASON_NONE ? res : CURLE_PARTIAL_FILE), mediaType);
+
+						if( res == CURLE_OPERATION_TIMEDOUT )
```
I feel this block could be optimized

Vinish100 commented on test/utests/tests/CurlTimeOutFailureReasonTests/CMakeLists.txt:
```diff
@@ -0,0 +1,82 @@
+# If not stated otherwise in this file or this component's license file the
+# following copyright and licenses apply:
+#
+# Copyright 2023 RDK Management
```
2025

Vinish100 commented on test/utests/tests/CurlTimeOutFailureReasonTests/CurlTimeOutFailureReasonTest.cpp:
```diff
@@ -0,0 +1,30 @@
+ * If not stated otherwise in this file or this component's license file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2023 RDK Management
```
2025

Vinish100 commented on test/utests/tests/CurlTimeOutFailureReasonTests/GetCurlTimeOutFailureReasonTest.cpp:
```diff
@@ -0,0 +1,289 @@
+* If not stated otherwise in this file or this component's license file the
+* following copyright and licenses apply:
+*
+* Copyright 2023 RDK Management
```
2025

Vinish100 commented on test/utests/tests/CurlTimeOutFailureReasonTests/GetCurlTimeOutFailureReasonTest.cpp:
```diff
@@ -0,0 +1,289 @@
+	 * @param[out] buffer Buffer containing manifest data
+	 * @retval true on success
+	*/
+	ManifestDownloadResponsePtr GetManifestForMPDDownloaderForDNSTimeout()
```
Lot of code duplication. Can't we just use a single function with multiple args?

Vinish100 commented on AampUtils.h:
```diff
@@ -422,4 +424,8 @@ int aamp_SetThreadSchedulingParameters(int policy, int priority);
  */
 bool aamp_isTuneScheme( const char *cmdBuf );
 
+CurlTimeoutFailureReason GetCurlTimeoutFailureReason(CURL* curl);
```
Add function doxygen header for new functions

lashmintha replied:
Added

narenr94 committed:
VPLAY-11457 [VIPA][LLAMA G1/G2/IT]: Technical Fault issue seen randomly while playing HD/UHD linear and VOD

VPLAY-11457 [VIPA][LLAMA G1/G2/IT]: Technical Fault issue seen randomly while playing HD/UHD linear and VOD

Reason for Change: Firebolt DRM request was sometimes (barely) timing out and failing.  Changed timeout from 3s to more generous 5s.

Test Guidance: monitor for reduced occurrence of VIPA tune failures

Risk: Low

Co-authored-by: nrames759 <Naren_Ramesh@comcast.com>

pstroffolino committed:
Reason for Change: replaced success macro with function
added doxygen comments for GetCurlTimeoutFailureReason
minor refactoring for GetCurlTimeoutFailureReason
removed few lines of commented out code

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Reason for Change: fix l1 tests on osx

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-10563_v2

pstroffolino committed:
Reason for Change: misc cleanup

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Reason for Change: cleanup

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>
